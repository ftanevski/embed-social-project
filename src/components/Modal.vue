<template>
    <div 
        class="modal in modal-bg" 
        data-testid="modal-bg-test"
        @click.self="closeModal"
    >
        <div class="modal-dialog">
            <div class="modal-content" data-testid="modal-content-test">
                <div class="block remove-margin-b">
                    <div class="block-header bg-primary-darker">
                        <ul class="block-options">
                            <li>
                                <button
                                    data-dismiss="modal"
                                    type="button"
                                    @click="closeModal"
                                    data-testid="times-button-test"
                                >
                                    <i class="fa fa-times"></i>
                                </button>
                            </li>
                        </ul>
                        <h3>
                            <slot name="header"></slot>
                        </h3>
                    </div>
                    <div class="block-content">
                        <slot name="text"></slot>
                    </div>
                </div>
                <div class="modal-footer">
                    <slot name="footer"></slot>
                </div>
            </div>
        </div>
    </div>
</template>
 
<script>
export default {
    name: 'Modal',
    emits: [
        'close-modal'
    ],
    methods: {
        closeModal() {
            this.$emit('close-modal');
        }
    }
}
</script>
 
<style scoped>
.modal-bg {
    background-color: rgba(0, 0, 0, 0.4);
    display: block;
}
.modal-dialog {
    top: 30%;
}
</style>