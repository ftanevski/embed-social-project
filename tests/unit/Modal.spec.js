import { mount } from '@vue/test-utils';
import Modal from '../../src/components/Modal';

describe('Modal', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(Modal, {
            slots: {
                header: 'Lorem Ipsum',
                content: '<p>Lorem ipsum dolor sit amet.</p>'
            }
        });
    });

    it('Matches the snapshot.', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Emits the closeModal to the parent component when clicking the X button.', async () => {
        let timesButton = wrapper.find('[data-testid="times-button-test"]');
        await timesButton.trigger('click');

        let emittedEvents = wrapper.emitted('close-modal');
        expect(emittedEvents).toHaveLength(1);
    });

    it('Emits the closeModal to the parent component when clicking outside of the modal.', async () => {
        let modalBg = wrapper.find('[data-testid="modal-bg-test"]');
        await modalBg.trigger('click');
        
        let emittedEvents = wrapper.emitted('close-modal');
        expect(emittedEvents).toHaveLength(1);
    });

    it('Doesn\'t emit anything when clicking inside of the modal.', async () => {
        let modalContent = wrapper.find('[data-testid="modal-content-test"]');
    
        await modalContent.trigger('click');
        let emittedEvents = wrapper.emitted('close-modal');
        expect(emittedEvents).toBeUndefined();
    });

    it('Renders the slot content', () => {
        expect(wrapper.html()).toContain('Lorem Ipsum');
        expect(wrapper.html()).toContain('<p>Lorem ipsum dolor sit amet.</p>');
    });
});