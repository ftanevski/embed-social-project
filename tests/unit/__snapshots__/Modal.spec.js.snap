// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Matches the snapshot. 1`] = `
<div
  class="modal in modal-bg"
  data-testid="modal-bg-test"
>
  <div
    class="modal-dialog modal-card"
  >
    <div
      class="modal-content"
      data-testid="modal-content-test"
    >
      <div
        class="block remove-margin-b"
      >
        <div
          class="block-header bg-primary-darker"
        >
          <ul
            class="block-options"
          >
            <li>
              <button
                data-dismiss="modal"
                data-testid="times-button-test"
                type="button"
              >
                <i
                  class="fa fa-times"
                />
              </button>
            </li>
          </ul>
        </div>
        <div
          class="block-content"
        >
          
          Lorem Ipsum
          
          
          <p>
            Lorem ipsum dolor sit amet.
          </p>
          
        </div>
      </div>
    </div>
  </div>
</div>
`;
